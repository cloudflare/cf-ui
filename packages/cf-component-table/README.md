# cf-component-table

> Cloudflare Table Component

## Installation

```sh
$ npm install cf-component-table
```

## Usage

```js
const React = require('react');
const {
  Table,
  TableHead,
  TableBody,
  TableFoot,
  TableRow,
  TableHeadCell,
  TableCell
} = require('../../src/index');

export default class Application extends React.Component {
  render() {
    const data = [
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9]
    ];

    return (
      <div>
        <h1>Cloudflare Table Component Example</h1>
        <Table>
          <TableHead>
            <TableRow>
              <TableHeadCell>One</TableHeadCell>
              <TableHeadCell>Two</TableHeadCell>
              <TableHeadCell>Three</TableHeadCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {data.map(row =>
              <TableRow>
                {row.map(cell =>
                  <TableCell>{cell}</TableCell>
                )}
              </TableRow>
            )}
          </TableBody>
        </Table>
      </div>
    );
  }
}
```
