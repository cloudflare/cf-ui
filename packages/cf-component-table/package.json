{
  "name": "cf-component-table",
  "description": "Cloudflare Table Component",
  "version": "2.4.1",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "author": "James Kyle <jkyle@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "dependencies": {
    "cf-style-container": "^1.3.1",
    "prop-types": "^15.5.8"
  },
  "peerDependencies": {
    "react": "^0.14.9 || ^15.0.0-0 || ^16.0.0-0"
  },
  "devDependencies": {
    "cf-style-provider": "^2.0.0"
  }
}
