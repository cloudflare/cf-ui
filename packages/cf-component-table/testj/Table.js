import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from 'cf-component-table';

describe('Table', () => {
  test('should render', () => {
    const component = renderer.create(<Table>Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render extra class names', () => {
    const component = renderer.create(<Table className="extra">Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render striped', () => {
    const component = renderer.create(<Table striped>Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render hover', () => {
    const component = renderer.create(<Table hover>Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render not bordered', () => {
    const component = renderer.create(<Table bordered={false}>Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render condensed', () => {
    const component = renderer.create(<Table condensed>Table</Table>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
