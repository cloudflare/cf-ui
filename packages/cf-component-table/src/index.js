import Table, { createTable } from './Table';
import TableBody, { createTableBody } from './TableBody';
import TableCell, { createTableCell } from './TableCell';
import TableFoot, { createTableFoot } from './TableFoot';
import TableHead, { createTableHead } from './TableHead';
import TableHeadCell, { createTableHeadCell } from './TableHeadCell';
import TableRow, { createTableRow } from './TableRow';
import TablePropTypes from './TablePropTypes';

export {
  Table,
  TableBody,
  TableCell,
  TableFoot,
  TableHead,
  TableHeadCell,
  TableRow,
  createTable,
  createTableBody,
  createTableCell,
  createTableFoot,
  createTableHead,
  createTableHeadCell,
  createTableRow,
  TablePropTypes
};
