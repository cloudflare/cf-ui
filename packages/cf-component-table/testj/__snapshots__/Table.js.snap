// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should render 1`] = `
<table
  className="cf-table cf-table--bordered"
>
  Table
</table>
`;

exports[`Table should render condensed 1`] = `
<table
  className="cf-table cf-table--bordered cf-table--condensed"
>
  Table
</table>
`;

exports[`Table should render extra class names 1`] = `
<table
  className="cf-table cf-table--bordered extra"
>
  Table
</table>
`;

exports[`Table should render hover 1`] = `
<table
  className="cf-table cf-table--hover cf-table--bordered"
>
  Table
</table>
`;

exports[`Table should render not bordered 1`] = `
<table
  className="cf-table"
>
  Table
</table>
`;

exports[`Table should render striped 1`] = `
<table
  className="cf-table cf-table--striped cf-table--bordered"
>
  Table
</table>
`;
